### PR TITLE
Update C++ bindings

### PR DESCRIPTION
**Pull request by mplooster**
Includes
* Improvements to CMake, which will allow the SDK to build on Windows (tested with Visual Studio 2019)
* Enable specifying SSL options when creating a C++ client.
* Update web docs to reflect these changes.

### CODE REVIEW

File: docs-src/content/connecting/_index.md [hunk 1]
```diff
@@ -46,22 +46,27 @@ func main() {
...
+int main(int argc, char* argv[])
+{
+  std::string serverAddress = "127.0.0.1:9002";
 
-#include <iostream>
-#include <string>
+  // Read or embed the digital certificate, and store
+  // in rootCert. Unlike other languages, in C++ there
+  // is nothing to do this automatically.
+  grpc::string rootCert;
```
Comment (happyalu) [inline docs-src/content/connecting/_index.md:59]: should we have some notes here for where to read this from on linux/windows?

Comment (mplooster) [reply docs-src/content/connecting/_index.md:59]: Well, TBH, I'm not sure where to read it from on Linux. On Windows, you have to use the Windows API and get it from the [certificate store](https://docs.microsoft.com/en-us/windows/win32/seccrypto/using-certificates). And what about Mac? Do we want to document where to get it from for all major platforms?

**Review by happyalu — APPROVED** [review]
Lgtm.

**Review by renaissanceGeek — APPROVED** [review]
I'm trusting Alok's approval for the cmake changes.  Everything else looks good to me! :-)

**Pull request merged**